### ISSUE TITLE
Cnes report is executed too soon in the Sonar Scanner stage

### ISSUE THREAD

**Issue by braisvq1996** [opendevstack/ods-jenkins-shared-library#732]
**Describe the bug**
After the Sonarqube scan ends successfully, cnes report is executed but if the backgound task in SonarQube have not ended the information picked would be from the previous scanner or if the branch is new, then the pipeline will fail.

**To Reproduce**
1. Create a new branch in Bitbucket
2. Execute a pipeline with Sonar scan enabled
3. Check the time of the execution

**Expected behavior**
The cnes report should be executed after the Sonarqube Background task has finished

**Screenshots**

Issue we found
<img width="722" alt="Captura2" src="https://user-images.githubusercontent.com/26645694/135991137-c262a907-95ff-423b-be49-9e1adbaaa7aa.PNG">

Execution timeline
<img width="586" alt="Captura3" src="https://user-images.githubusercontent.com/26645694/135991235-3215115d-ee0d-4b81-8398-5e417c10e1df.PNG">

**Affected version (please complete the following information):**
 - OpenShift: 3.11
 - OpenDevStack 3.1

**Log Output (ensure to remove any confidential information like tokens, project names, etc.**
```
12:49:02.629 DEBUG: Post-jobs : 
12:49:02.930 INFO: Analysis total time: 1:46.421 s
12:49:02.932 INFO: ------------------------------------------------------------------------
12:49:02.932 INFO: EXECUTION SUCCESS
12:49:02.932 INFO: ------------------------------------------------------------------------
12:49:02.932 INFO: Total time: 1:51.673s
12:49:03.019 INFO: Final Memory: 14M/50M
12:49:03.019 INFO: ------------------------------------------------------------------------
[Pipeline] }
[Pipeline] // withSonarQubeEnv
[Pipeline] echo
DEBUG: [key-client-sq-scan]  (took 115043 ms)
[Pipeline] withSonarQubeEnv
Injecting SonarQube environment variables using the configuration: SonarServerConfig
[Pipeline] {
[Pipeline] sh (Generate CNES Report)
+ java -jar /usr/local/cnes/cnesreport.jar -s https://sonarqube-ods.XXXX.com -t ****** -p key-client -a jenkins-key-cd-key-cd-cdbmgr-mro-pipeline-master-59 -b release/0.0.2
SonarQube URL: https://sonarqube-ods.XXXX.com
SonarQube online: true
Detected SonarQube version: 8.2.0
[ERROR] Not found error sent by SonarQube server (code 404, URL https://sonarqube-ods.XXXX.com/api/navigation/component?component=key-client&branch=release/0.0.2, Error {"errors":[{"msg":"Component 'key-client' on branch 'release/0.0.2' not found"}]}), please check cnesreport compatibility with your SonarQube server version.
[Pipeline] }
[Pipeline] // withSonarQubeEnv
[Pipeline] }
[Pipeline] // stage
[Pipeline] echo
[SonarQube Analysis-client] **** ENDED stage 'SonarQube Analysis' for component 'client' branch 'release/0.0.2' **** (took 118356 ms)
```

**Additional context**
It took 3 seconds after the Sonar scan cli has finished and the error appeared in the pipeline (cnes report), and checking the background task for this issue, it took 10 seconds to finish.
So if the cnes report was executed 15 seconds later, it would not have failed.
Checking the history of the time it takes the background tasks to finish, I can see diferent times, from miliseconds to 20 seconds

![image](https://user-images.githubusercontent.com/26645694/135992444-acefa014-aa45-4f00-a895-d854ab5c06c0.png)

**Comment by braisvq1996** [comment]
Should we set a sleep between the sonar scanner cli and cnes report ?

**Comment by michaelsauter** [comment]
Nice one :(

Maybe we can use the API to check if the task finished?

**Comment by braisvq1996** [comment]
Yeah, already working on that, I am planing to use this API endpoint: api/ce/task
![image](https://user-images.githubusercontent.com/26645694/136019757-8ad191a7-c724-4316-99cf-73bcc4ab9df2.png)
